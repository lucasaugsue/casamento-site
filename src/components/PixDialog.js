import PixIcon from '@mui/icons-material/Pix';
import { Dialog, Grid, Slide } from '@mui/material';
import Typography from '@mui/material/Typography';
import React from 'react';
import styles from './PixDialog.module.css';


const Transition = React.forwardRef(function Transition(props, ref) {
	return <Slide direction="up" ref={ref} {...props} />;
});

export default function PixDialog({open, handleClose}){

    return <div>
        <Dialog
            open={open ? open : false}
            maxWidth="lg"
            onClose={() => handleClose()}
            TransitionComponent={Transition}
        >
            <Grid 
                container
                spacing={1}
                alignContent="center"
                flexDirection="column"
                alignItems="center"
                className={styles.dialogContent}
            >
                <Grid item xs={12} md={12}>
                    <div className={styles.pague}>Pague com</div>
                </Grid>
                <Grid item xs={12} md={12}>
                    <Grid 
                        container
                        spacing={1}
                        alignContent="center"
                        flexDirection="row"
                        alignItems="center"
                    >
                        <PixIcon style={{fontSize: "10vh"}}/>
                        <Typography style={{
                            fontWeight: "600",
                            margin: "0 0 0 1vw",    
                            fontSize: '9vh'
                        }}> Pix </Typography>
                    </Grid>
                </Grid>

                <div
                    loading="lazy"
                    className={styles.pixImage}
                />

                <div className={styles.sePreferir}>Se preferir</div>
                <div className={styles.nossaChave}>Nossa chave</div>
                
                <div className={styles.chavePix}>079.861.121.90</div>
                <div className={styles.nome}>No nome de Lucas AugsuÃª</div>
            </Grid>
        </Dialog>
    </div>
}
