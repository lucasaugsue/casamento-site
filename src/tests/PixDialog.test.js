import renderer from 'react-test-renderer';
import PixDialog from '../components/PixDialog';

// Define o mock para o Dialog
jest.mock('@mui/material/Dialog');

test('Renderiza PixDialog corretamente', async () => {
    const handleClose = jest.fn();

    const tree = renderer
	.create(<PixDialog
        open={true} 
        handleClose={() => handleClose()} 
	/>)
	.toJSON();
    expect(tree).toMatchSnapshot();
});