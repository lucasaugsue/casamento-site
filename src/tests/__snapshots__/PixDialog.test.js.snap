// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderiza PixDialog corretamente 1`] = `<div />`;
